modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverSetUtils.java

```diff
@@ -188,82 +203,114 @@ private static ValueResolver getParameterValueResolver(String parameterName, Met
+                                         valueResolverFactory));
+          } catch (MuleException e) {
+            throw new MuleRuntimeException(e);
           }
-          resolverReference
-              .set(getValueResolverFor(parameterName, objectType, resolveAndinjectIfStatic(resolver, muleContext),
-                                       getDefaultValue(type),
-                                       getExpressionSupport(objectType), false, modelProperties, actualAcceptReferences));
-        } catch (MuleException e) {
-          throw new MuleRuntimeException(e);
         }
-      }
 
-      @Override
-      protected void defaultVisit(MetadataType metadataType) {
-        resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
-                                                  getExpressionSupport(metadataType), false, modelProperties, acceptsReferences));
-      }
+        @Override
+        protected void defaultVisit(MetadataType metadataType) {
+          resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
+                                                    getExpressionSupport(metadataType), false, modelProperties,
+                                                    acceptsReferences, valueResolverFactory));
+        }
 
-      private ValueResolver getValueResolverFor(String parameterName, MetadataType metadataType, Object value,
-                                                Object defaultValue, ExpressionSupport expressionSupport, boolean required,
-                                                Set<ModelProperty> modelProperties, boolean acceptsReferences) {
-        return valueResolverFactory.of(parameterName, metadataType, value, defaultValue,
-                                       expressionSupport, required, modelProperties, acceptsReferences);
-      }
 
-    });
+
+      });
+    } else {
+      Object convertedValue = convertValueWithExpressionLanguage(value, type, muleContext.getExpressionManager());
+      resolverReference.set(getValueResolverFor(parameterName, type, convertedValue, null, NOT_SUPPORTED, false, modelProperties,
+                                                false, valueResolverFactory));
+    }
     initialiseIfNeeded(resolverReference.get(), muleContext);
     return resolverReference.get();
   }
 
+  private static Object convertValueWithExpressionLanguage(Object value, MetadataType type, ExpressionManager expressionManager) {
+    return expressionManager.evaluate(PAYLOAD_EXPRESSION,
+                                      DataType.builder().type(getType(type).orElse(Object.class))
+                                          .mediaType(getFirstValidMimeType(type))
+                                          .build(),
+                                      BindingContext.builder().addBinding(PAYLOAD, TypedValue.of(value)).build())
```

Comment:
when you make `TypedValue.of(value)`  you're defaulting to `application/java` as the mimeType. We will always need a version of this that receives `Object`, but should there be an overload  in which `value` is already supplied as a TypedValue?